code/modules/shuttle/on_move.dm

```diff
@@ -11,6 +11,28 @@
 		return 0
 	. = ..()
 
+/obj/machinery/atmospherics/onShuttleMove()
+	. = ..()
+	for(DEVICE_TYPE_LOOP)
+		if(get_area(nodes[I]) != get_area(src))
+			nullifyNode(I)
+
+/obj/structure/cable/onShuttleMove()
+	. = ..()
+	var/A = get_area(src)
+	//cut cables on the edge
+#define DIR_CHECK_TURF_AREA(X) (get_ranged_target_turf(src, X, 1) != A)
+	if(DIR_CHECK_TURF_AREA(NORTH) || DIR_CHECK_TURF_AREA(SOUTH) || DIR_CHECK_TURF_AREA(EAST) || DIR_CHECK_TURF_AREA(WEST))
+		cut_cable_from_powernet()
+
+/obj/structure/disposalpipe/onShuttleMove()
+	. = ..()
+	var/A = get_area(src)
+	//break pipes on the edge
+	if(DIR_CHECK_TURF_AREA(NORTH) || DIR_CHECK_TURF_AREA(SOUTH) || DIR_CHECK_TURF_AREA(EAST) || DIR_CHECK_TURF_AREA(WEST))
+		deconstruct(FALSE)
```

Comment:
Argument 1: Being forcefully ripped off a brother pipe should break it
Argument 2: Transit space entry is rough on things, mmkay?